File: FaucetPay .Net FrameWork/Models/CheckAddress.vb
Imports FaucetPay.FaucetPayPost
Imports FaucetPay.FaucetPayLogs
Imports System.Text.Json

Namespace Models
    Friend Class CheckAddress

        Private Class CheckAddressItems
            Public Property status As Int32
            Public Property message As String
            Public Property payout_user_hash As String
        End Class

        Friend Function CheckHashPayout(API As String, Address As String, ByVal Logs As Boolean, Optional Currency As String = "BTC") As String
            Dim Request As String = Nothing
            Try
                Dim Post = New PostRequest()
                Dim Site As String = "https://faucetpay.io/api/v1/checkaddress"
                Request = Post.SEND(Site, "&api_key=" & API & "&address=" & Address & "&currency=" & Currency, PostRequest.Method.POST_, PostRequest.Secure.TLS12)
                Dim Deserealized = JsonSerializer.Deserialize(Of CheckAddressItems)(Request)

                If Deserealized.status <> 200 Then
                    Dim LOG As New FLog
                    If Logs = True Then
                        LOG.Loggs("Error: " & Deserealized.status & " " & Deserealized.message)
                    End If
                    Return "Error: " & Deserealized.status & " " & Deserealized.message
                End If

                If Logs = True Then
                    Dim LOG As New FLog
                    LOG.Loggs("Payout User Hash: " & Deserealized.payout_user_hash)
                End If
                Return Deserealized.payout_user_hash

            Catch ex As Exception
                Dim Deserealized = JsonSerializer.Deserialize(Of CheckAddressItems)(Request)
                Return Deserealized.message
            End Try
        End Function

        Friend Function CheckAddressExist(API As String, Address As String, ByVal Logs As Boolean, Optional Currency As String = "BTC") As Boolean
            Dim Request As String = Nothing
            Try
                Dim Post = New PostRequest()
                Dim Site As String = "https://faucetpay.io/api/v1/checkaddress"
                Request = Post.SEND(Site, "&api_key=" & API & "&address=" & Address & "&currency=" & Currency, PostRequest.Method.POST_, PostRequest.Secure.TLS12)
                Dim Deserealized = JsonSerializer.Deserialize(Of CheckAddressItems)(Request)


                If Deserealized.status <> 200 Then
                    Dim LOG As New FLog
                    If Logs = True Then
                        LOG.Loggs("Ckeck Address Exist: " & Address & " is False")
                    End If
                    Return False
                Else
                    If Logs = True Then
                        Dim LOG As New FLog
                        LOG.Loggs("Ckeck Address Exist: " & Address & " is True")
                    End If
                    Return True
                End If

            Catch ex As Exception
                Return False
            End Try
        End Function
    End Class
End Namespace

File: FaucetPay CSharp .Net FrameWork/Models/CheckAddress.cs
using System;
using System.Text.Json;
using FaucetPayLogs;
using FaucetPayPost;

namespace Models
{
    internal partial class CheckAddress
    {

        private partial class CheckAddressItems
        {
            public int status { get; set; }
            public string message { get; set; }
            public string payout_user_hash { get; set; }
        }

        internal string CheckHashPayout(string API, string Address, bool Logs, string Currency = "BTC")
        {
            string Request = null;
            try
            {
                var Post = new PostRequest();
                string Site = "https://faucetpay.io/api/v1/checkaddress";
                Request = Post.SEND(Site, "&api_key=" + API + "&address=" + Address + "&currency=" + Currency, PostRequest.Method.POST_, PostRequest.Secure.TLS12);
                var Deserealized = JsonSerializer.Deserialize<CheckAddressItems>(Request);

                if (Deserealized.status != 200)
                {
                    var LOG = new FLog();
                    if (Logs == true)
                    {
                        LOG.Loggs("Error: " + Deserealized.status + " " + Deserealized.message);
                    }
                    return "Error: " + Deserealized.status + " " + Deserealized.message;
                }

                if (Logs == true)
                {
                    var LOG = new FLog();
                    LOG.Loggs("Payout User Hash: " + Deserealized.payout_user_hash);
                }
                return Deserealized.payout_user_hash;
            }

            catch 
            {
                var Deserealized = JsonSerializer.Deserialize<CheckAddressItems>(Request);
                return Deserealized.message;
            }
        }

        internal bool CheckAddressExist(string API, string Address, bool Logs, string Currency = "BTC")
        {
            string Request = null;
            try
            {
                var Post = new PostRequest();
                string Site = "https://faucetpay.io/api/v1/checkaddress";
                Request = Post.SEND(Site, "&api_key=" + API + "&address=" + Address + "&currency=" + Currency, PostRequest.Method.POST_, PostRequest.Secure.TLS12);
                var Deserealized = JsonSerializer.Deserialize<CheckAddressItems>(Request);


                if (Deserealized.status != 200)
                {
                    var LOG = new FLog();
                    if (Logs == true)
                    {
                        LOG.Loggs("Ckeck Address Exist: " + Address + " is False");
                    }
                    return false;
                }
                else
                {
                    if (Logs == true)
                    {
                        var LOG = new FLog();
                        LOG.Loggs("Ckeck Address Exist: " + Address + " is True");
                    }
                    return true;
                }
            }

            catch 
            {
                return false;
            }
        }
    }
}